nautobot/extras/tables.py

```diff
@@ -919,28 +920,36 @@ def render_summary(self, record):
 
     class Meta(BaseTable.Meta):
         model = JobResult
-        fields = (
-            "pk",
-            "date_created",
-            "name",
-            "job_model",
-            "scheduled_job",
-            "duration",
-            "date_done",
-            "user",
-            "status",
-            "summary",
-            "actions",
+        fields = tuple(
+            field
+            for field in [
+                "pk",
+                "date_created",
+                "name",
+                "job_model",
+                "scheduled_job",
+                "duration",
+                "date_done",
+                "user",
+                "status",
+                "summary" if get_settings_or_config("JOB_RESULTS_SUMMARY_FIELD_ENABLED") else None,
```

Comment:
As you mentioned on Slack, this probably won't be dynamic as written since this code is evaluated at import time. Probably disabling this column should be done instead in the `JobResultListView.extra_context()` method (by taking the `"table"` context entry provided by the base view, and calling `table.columns.hide("summary")` if necessary) since that gets evaluated every time the view is rendered.